src/hooks/WorkspaceDatasetsHooks.js

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import { useMemo } from 'react';
+import { useDatasets } from '../state/hooks/DatasetHooks';
+import { useWorkspaceDatasetsFilter } from '../state/hooks/WorkspaceHooks';
+
+export const useWorkspaceDatasets = () => {
+  const datasets = useDatasets();
+  const workspaceDatasetsFilter = useWorkspaceDatasetsFilter();
+
+  return useMemo(() => {
+    const getMainDatasets = () => datasets.filter((dataset) => dataset.main === true);
+    if (!workspaceDatasetsFilter) return getMainDatasets();
+
+    const workspaceDatasets = [];
+    workspaceDatasetsFilter.forEach((filterItem) => {
+      if (typeof filterItem !== 'string')
+        console.warn(`Ignoring dataset filter entry ${filterItem} because it is not a string`);
+      else {
+        const filteredDataset = datasets.find((dset) => dset.id === filterItem);
+        if (!filteredDataset) console.warn(`No dataset found for filter entry ${filterItem}`);
+        else workspaceDatasets.push(filteredDataset);
+      }
+    });
+
+    if (workspaceDatasetsFilter.length > 0 && workspaceDatasets.length === 0) {
```

Comment:
No it isn't, because an empty filter list is a valid use case, we don't want to log a warning in this case, even though the results list will be empty